### ISSUE TITLE
ApolloError not thrown when receiving mutation result with GraphQL errors

### ISSUE THREAD

**Issue by hammadj** [apollostack/apollo-client#458]
ApolloError is not thrown when the result of a mutation contains errors. Haven't tested with queries yet so I don't know if this is only a problem for mutations.

Example:
```js
client.mutate({
  mutation: gql`
    mutation update($input: ProfileUpdateType) {
      updateUser(input: $input) {
        _id
        email
        name
        picture
      }
    }
  `,
  variables: {
    input: {
      invalidField: 'value'
    }
  }
})
.then(({ errors }) => {
  if (errors) {
    console.log('ApolloError not thrown')
  }
})
.catch(error => console.log('ApolloError: ', error.message));
```

Results in `ApolloError not thrown` being logged to the console.

Example of result with errors:
![image](https://cloud.githubusercontent.com/assets/7143583/17080241/a4070b48-50dc-11e6-8c16-c017a491e4ce.png)

**Comment by Poincare** [comment]
#465 should be the fix for this.

**Comment by stubailo** [comment]
Released in 0.4.8!
